### ISSUE TITLE
Avoid to overwrite existing Labels if none specified

### ISSUE THREAD

**Issue by toninofox** [google-github-actions/deploy-cloudrun#429]
### TL;DR

Avoid to overwrite existing labels

### Detailed design

A normal process may involve to create cloud run skeleton using Terraform with all metadata and labels on it.
The actual cloud run code / image will then be deployed by another github repository workflow using the deploy-cloud run action (parametric deployment without a template.yml).

as per documentation:

> labels: (Optional) List of key=value pairs to set as labels on the Cloud Run service. **Existing labels will be overwritten.**
> 
> with:
>   labels:
>     my-label=my-value
> The GitHub Action **will automatically apply the following labels** which Cloud Run uses to enhance the user experience:
> 
> managed-by: github-actions
> commit-sha: <sha>
> 

Because these automatic labels are always applied by the github action, the original labels are being overwritten.

It would be nice to disable the automatic labels on demand (to avoid changing the current default behaviour) so they won't overwrite the existing ones.

**Comment by sethvargo** [comment]
Hi @toninofox 

Because GitHub Actions does not have information about prior Cloud Run state, we cannot "append-only" labels. We would be unable to articulate when a label is _removed_ if we treat the action.yml as imperative instead of declarative. As such, we've made the decision to make the action.yml declarative. You can find old issues from when this action used to be imperative, and it fails to meet multiple critical user journeys.

We can add a setting to skip applying the default labels, but I don't think it's feasible to change the behavior to be imperative.

**Comment by toninofox** [comment]
Hi @sethvargo thanks for the quick reply.
The replacement of existing labels whenever someone defines new ones makes sense to me and I am not asking in changing this behaviour.
The problem is when I don't define any **custom** label to be applied during my cloud run deployment.

The expected behaviour when the field **labels** is empty during the deployment stage would be to keep the old/existing ones defined in the cloud run.
However, even when the label field is empty, the "unwanted" automatic labels applied by the github action (managed by and commit-sha), are always being applied and they overwrite the old ones even.

example of expected behaviour:
- Deploy a cloud RUN with the `deploy-cloudrun` action with no labels
- Manually apply a custom label on the cloud run
- deploy the cloud run with the `deploy-cloudrun` action with no labels
- the custom label still appears in the yaml

Hope that makes sense but feel free if any further detail is needed
